File: backup/entities/Driver.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'drivers')]
class Driver
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\Column(name: 'userId', type: 'integer')]
    private ?int $userId = null;

    #[ORM\Column(name: 'vehicleType', type: 'string', length: 100)]
    private ?string $vehicleType = null;

    #[ORM\Column(name: 'licenseNumber', type: 'string', length: 50)]
    private ?string $licenseNumber = null;

    #[ORM\Column(name: 'availableStatus', type: 'boolean', options: ['default' => true])]
    private ?bool $availableStatus = true;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getUserId(): ?int
    {
        return $this->userId;
    }

    public function setUserId(int $userId): self
    {
        $this->userId = $userId;
        return $this;
    }

    public function getVehicleType(): ?string
    {
        return $this->vehicleType;
    }

    public function setVehicleType(string $vehicleType): self
    {
        $this->vehicleType = $vehicleType;
        return $this;
    }

    public function getLicenseNumber(): ?string
    {
        return $this->licenseNumber;
    }

    public function setLicenseNumber(string $licenseNumber): self
    {
        $this->licenseNumber = $licenseNumber;
        return $this;
    }

    public function isAvailableStatus(): ?bool
    {
        return $this->availableStatus;
    }

    public function setAvailableStatus(bool $availableStatus): self
    {
        $this->availableStatus = $availableStatus;
        return $this;
    }
}

File: src/main/java/Models/Driver.java
package Models;

public class Driver {
    private int id;
    private int userId; // Foreign key reference to User table
    private String vehicleType;
    private String licenseNumber;
    private boolean availableStatus;

    // Constructor
    public Driver(int id, int userId, String vehicleType, String licenseNumber, boolean availableStatus) {
        this.id = id;
        this.userId = userId;
        this.vehicleType = vehicleType;
        this.licenseNumber = licenseNumber;
        this.availableStatus = availableStatus;
    }

    // Getters and Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public String getVehicleType() {
        return vehicleType;
    }

    public void setVehicleType(String vehicleType) {
        this.vehicleType = vehicleType;
    }

    public String getLicenseNumber() {
        return licenseNumber;
    }

    public void setLicenseNumber(String licenseNumber) {
        this.licenseNumber = licenseNumber;
    }

    public boolean isAvailableStatus() {
        return availableStatus;
    }

    public void setAvailableStatus(boolean availableStatus) {
        this.availableStatus = availableStatus;
    }
}
